CHANGELOG

```diff
@@ -1,3 +1,55 @@
+  - Windows: Significantly improved the initial user experience with Windows platform. Fixed many
+    issues in File input. Added support for using the plugin framework (installing, upgrading, removing)
+  - With the release of Kibana 4, we have removed the `bin/logstash web` command and any reference to
+    Kibana from Logstash
+
+  # input
+  - Added new CouchDB input plugin to fetch data from CouchDB. Using the _changes API, data can be kept
+    in sync with any output like Elasticsearch by using this input.
+  - File Input: Fixed a number of issues on Windows platform. These include resolving file locking
+    issues which was causing log files to not rotate. Also added support for using SinceDB to record
+    multiple files' last read information. Also fixed encoding issues which applies to many inputs (#2507). 
+  - Kafka: Added support for whitelisting and blacklisting topics in the input. 
+  - EventLog: For Windows, this input gracefully shutsdown if there is a timeout while receiving events
+  - Heartbeat: We created a new input plugin for generating heartbeat messages at periodic intervals. 
+    Use this to monitor Logstash -- you can measure the latency of the pipeline using these heartbeat events,
+    and also check for availability
+  - S3: Added IAM roles support so you can securely read and write events from S3 without providing your
+    AWS credentials. Added support for using temporary credentials obtained from AWS STS
+  - Lumberjack: Resolved issue where unrelated events were getting merged into a single event while using
+    this input with with the multiline codec (#2016)
+
+  # filter
+  - Mutate: Resolved issue where convert option was creating an extra field in the event (#2268)
+  - Mutate: Fixed issue where mutate with non-existent field was throwing an error (#2379)
+  - Multiline: Handled cases where we dropped lines from certain multiline xml messages 
+
+  # output
+  - Elasticsearch: Added support to be more resilient to transient errors in Elasticsearch. Previously, partial
+    failures from the bulk indexing functionality were not handled properly. With this fix, we added the ability
+    to capture failed requests from Elasticsearch and retry them. Error codes like 429 (too many requests) will
+    now be retried by default for 3 times. The number of retries and the interval between consecutive retries 
+    can be configured
+  - S3: Fixed a critical problem in the S3 Output plugin when using the size_file option. This could cause data loss
+    and data corruption of old logs. Added IAM roles support so you can securely read and write events from S3 
+    without providing your AWS credentials. Added support for using temporary credentials obtained from AWS STS
+
+  # codec
+  - Added new Elasticsearch bulk codec which can be used to read data formatted in the Elasticsearch 
+    Bulk API (multiline json) format. For example, this codec can be used in combination with RabbitMQ input
```

Comment:
If anything can be added here, I'd say something about its functionality being like the Elasticsearch RabbitMQ river